File: frontend/src/types/curriculum.ts
export interface ChapterSummary {
  id: string | null;
  order_index: number;
  title: string;
  description: string;
  status: "locked" | "available" | "in_progress" | "completed";
}

export interface Curriculum {
  subject_id: string;
  subject_name: string;
  chapters: ChapterSummary[];
}

File: backend/app/schemas/curriculum.py
from pydantic import BaseModel


class ChapterSummary(BaseModel):
    id: str | None = None
    order_index: int
    title: str
    description: str
    status: str = "locked"


class CurriculumGenerateRequest(BaseModel):
    subject_name: str
    grade: str | None = None
    board: str | None = None  # e.g. "CBSE", "ICSE", "Cambridge IGCSE", "IB", "Common Core"
    background: str | None = None
    difficulty_level: str = "beginner"


class CurriculumResponse(BaseModel):
    subject_id: str
    subject_name: str
    chapters: list[ChapterSummary]
